src/upload_node.py

```diff
@@ -0,0 +1,117 @@
+from dataclasses import dataclass
+
+import rospy
+
+from sensor_msgs.msg import NavSatFix
+from std_msgs.msg import Float64
+
+from martian_mines.msg import FigureMsgList, FigureMsg
+from utils.coords_scaler import CoordinateScaler
+from utils.uploader import Uploader
+
+
+@dataclass
+class DroneData:
+    global_pose: NavSatFix = None
+    heading: Float64 = None
+    rel_altitude: Float64 = None
+
+    def is_complete(self) -> bool:
+        return all(param is not None for param in (self.global_pose, self.heading, self.rel_altitude))
+
+    def reset(self) -> None:
+        self.global_pose = None
+        self.heading = None
+        self.rel_altitude = None
+
+
+class UploadNode:
+    def __init__(self):
+        self.global_pose_sub = rospy.Subscriber("uav0/mavros/global_position/global", NavSatFix, self.global_pose_callback)
+        self.heading_sub = rospy.Subscriber("uav0/mavros/global_position/compass_hdg", Float64, self.heading_callback)
+        self.alt_sub = rospy.Subscriber("uav0/mavros/global_position/rel_alt", Float64, self.altitude_callback)
+        self.figure_sub = rospy.Subscriber("uav0/detection/confirmed_figures", FigureMsgList, self.figure_callback)
+        self.start_pose = DroneData()
+        self.drone_uploader = Uploader("http://91.227.41.24:3001/drones")
+        self.figure_uploader = Uploader("http://91.227.41.24:3001/samples")
+        self.drone_data = DroneData()
+        self.rate = rospy.Rate(10)
+        self.ready = False
+        self.figures = None
+
+    def run(self):
+        while True:
```

Comment:
Use `not rospy.is_shutdown()`  instead of while True

Reply:
not needed anymore, everything has moved to callbacks / __init__()